src/cla/tcp.rs

```diff
@@ -0,0 +1,511 @@
+                match Bundle::try_from(vec) {
+                    Ok(bundle) => {
+                        tokio::spawn(async move {
+                            if let Err(err) = crate::core::processing::receive(bundle).await {
+                                error!("Failed to process bundle: {}", err);
+                            }
+                        });
+                    }
+                    Err(err) => {
+                        error!("Failed to parse bundle: {}", err);
+                        self.sender_task
+                            .send(TcpClPacket::XferRefuse(XferRefuseData {
+                                reason: XferRefuseReasonCode::NotAcceptable,
+                                tid: data.tid,
+                            }))
+                            .await?;
+                    }
+                }
+                return Ok(connection_closing);
+            }
+            _ => {
+                error!("Unexpected packet");
+                return Ok(true);
+            }
+        }
+    }
+    async fn send(&mut self, data: (SocketAddr, Vec<u8>)) -> bool {
+        let mut byte_vec = Vec::new();
+        let mut acked = 0u64;
+        self.last_tid += 1;
+        let vec = data.1;
+        for bytes in vec.chunks(self.data_remote.segment_mru as usize) {
+            let buf = Bytes::copy_from_slice(bytes);
+            let len = buf.len() as u64;
+            debug!("bytes len {}", len);
+            let packet_data = XferSegData {
+                flags: XferSegmentFlags::empty(),
+                buf,
+                len,
+                tid: self.last_tid,
+            };
+            byte_vec.push(packet_data);
+        }
+        if byte_vec.is_empty() {
+            warn!("Emtpy bundle transfer, aborting");
+            return false;
+        }
+        byte_vec.first_mut().unwrap().flags |= XferSegmentFlags::START;
+        byte_vec.last_mut().unwrap().flags |= XferSegmentFlags::END;
+        for packet in byte_vec {
+            if let Err(err) = self.sender_task.send(TcpClPacket::XferSeg(packet)).await {
+                error!("Internal sender error, {}", err);
+            }
+        }
+        while acked < vec.len() as u64 {
+            if let Some(received) = self.receiver_task.recv().await {
+                match received {
+                    TcpClPacket::XferAck(ack_data) => {
+                        if ack_data.tid == self.last_tid {
+                            acked = ack_data.len;
+                        }
+                    }
+                    TcpClPacket::XferRefuse(refuse_data) => {
+                        warn!("Transfer refused, {:?}", refuse_data.reason);
+                        break;
+                    }
+                    TcpClPacket::MsgReject(msg_reject_data) => {
+                        warn!("Message rejected, {:?}", msg_reject_data.reason);
+                        break;
+                    }
+                    _ => warn!("Unexpected packet"),
```

Comment:
add a debug print giving a few more information, packet type, packet bytes, magic numbers, first 10 bytes or something to debug the issue